src/utils/daos_hdlr.c

```diff
@@ -583,13 +637,30 @@ cont_list_snaps_hdlr(struct cmd_args_s *ap)
 	if (expected_count < snaps_count)
 		fprintf(stderr, "size required to gather all snapshots has raised, list has been truncated\n");
 
-	for (i = 0; i < min(expected_count, snaps_count); i++)
-		D_PRINT(DF_U64" ", buf[i]);
-	D_PRINT("\n");
+	if (snapname == NULL) {
+		for (i = 0; i < min(expected_count, snaps_count); i++)
+			D_PRINT(DF_U64" %s\n", epochs[i], names[i]);
+	} else {
+		for (i = 0; i < min(expected_count, snaps_count); i++)
+			if (strcmp(snapname, names[i]) == 0) {
+				if (epoch != NULL)
+					*epoch = epochs[i];
+				break;
+			}
+		if (i == min(expected_count, snaps_count)) {
+			fprintf(stderr, "%s not found in snapshots list\n", snapname);
+			rc = -DER_NONEXIST;
+		}
+	}
 
 out:
-	if (buf != NULL)
-		D_FREE(buf);
+	if (epochs != NULL)
+		D_FREE(epochs);
```

Comment:
```suggestion
	D_FREE(epochs);
```

Reply:
That's fine, both free() and D_FREE() are OK to be called with NULL.  This was discussed some time ago and it's both easier to read the code if it's shorter and easier to follow the logging entries if you don't check for NULL before calling D_FREE().

Reply:
I can't believe it, free() is now handling NULL parameter after multiple decades !!??

Reply:
https://man7.org/linux/man-pages/man3/free.3.html

"If ptr is NULL, no operation is performed.".

Reply:
Ok, let say it is an improvement vs Unix/libc ;-)